mono/mini/debugger-agent.c

```diff
@@ -5167,8 +5203,10 @@ ss_start (SingleStepReq *ss_req, MonoMethod *method, SeqPoint* sp, MonoSeqPointI
 			for (i = 0; i < sp->next_len; i++) {
 				next_sp = &next[i];
 
-				bp = set_breakpoint (method, next_sp->il_offset, ss_req->req, NULL);
-				ss_req->bps = g_slist_append (ss_req->bps, bp);
+				if (!ss_bp_is_duplicate(ss_req->bps, method, next_sp->il_offset)) {
```

Comment:
It could happen with the seq points at the beginning of a huge switch statement for example.
We might want to fall back to a hash table for entries > 10 for example.